adr/query.py

```diff
@@ -128,6 +129,30 @@ def run_query(name, args):
 
     logger.trace(f"JSON representation of query:\n{query_str}")
     result = query_activedata(query_str, config.url)
+
+    if result.get('url'):
+        # We must wait for the content
+        problem = 0
+        while problem < 3:
```

Comment:
`requests` has a built-in retry mechanism that we should use instead of rolling our own. Peterbe (from Mozilla) wrote a good post on it here:
https://www.peterbe.com/plog/best-practice-with-retries-with-requests

You can skip down to the `The Solution` if you just want to see the implementation + usage.

Reply:
I'm going to go ahead and make these changes as Marco would like them asap.

Reply:
Sorry, looking at the code I guess I misinterpreted what this was doing. It's retrying the `monitor` URL status and not retrying the `big results` url.

Though I'm still a bit worried that this could result in an infinite loop if the monitor URL never gets written.

Reply:
Yes, an infinite loop is possible if there is an uncontrolled shutdown of the ActiveData process.  

@ahal Please review the ActiveData side to ensure it is doing its best:

https://github.com/mozilla/ActiveData/blob/dev/vendor/jx_elasticsearch/es52/set_bulk.py
https://github.com/mozilla/ActiveData/blob/dev/vendor/jx_elasticsearch/es52/agg_bulk.py